Sources/Core/JavaModelRenderer.swift

```diff
@@ -168,6 +172,83 @@ public struct JavaModelRenderer: JavaFileRenderer {
         return [props, [bits]]
     }
 
+    // MARK: - TypeAdapterFactory
+
+    func renderTypeAdapterFactoryMethods() -> [JavaIR.Method] {
+        return [JavaIR.method(annotations: ["Override"], [.public], "<T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken)") { [
+            JavaIR.ifBlock(condition: "!" + className + ".class.isAssignableFrom(typeToken.getRawType())") { [
+                "return null;",
+            ] },
+            "return (TypeAdapter<T>) new " + className + "TypeAdapter(gson, this, typeToken);",
+        ] }]
+    }
+
+    // MARK: - TypeAdapter
+
+    func renderTypeAdapterProperties() -> [[JavaIR.Property]] {
+        let delegate = JavaIR.Property(
+            annotations: [],
+            modifiers: [.final, .private],
+            type: "TypeAdapter<" + className + ">",
+            name: "delegateTypeAdapter",
+            initialValue: ""
+        )
+
+        let elementTypeAdapter = JavaIR.Property(
+            annotations: [],
+            modifiers: [.final, .private],
+            type: "TypeAdapter<JsonElement>",
+            name: "elementTypeAdapter",
+            initialValue: ""
+        )
+
+        return [[delegate, elementTypeAdapter]]
+    }
+
+    func renderTypeAdapterMethods() -> [JavaIR.Method] {
+        let constructor = JavaIR.method(
+            annotations: [],
+            [.public],
+            className + "TypeAdapter(Gson gson, " + className + "TypeAdapterFactory factory, TypeToken typeToken)"
+        ) { [
+            "this.delegateTypeAdapter = gson.getDelegateAdapter(factory, typeToken);",
+            "this.elementTypeAdapter = gson.getAdapter(JsonElement.class);",
+        ] }
+
+        let write = JavaIR.method(
+            annotations: ["Override"],
+            [.public],
+            "void write(JsonWriter writer, " + className + " value) throws IOException"
+        ) { [
+            "this.delegateTypeAdapter.write(writer, value);",
+        ] }
+
+        let read = JavaIR.method(
+            annotations: ["Override"],
```

Comment:
Question: Should we create an enum represent the various valid annotations?

Reply:
I plan to add the ability for custom annotations to be passed in through a command-line option / auxillary schema file so I don't think I can restrict it to an enum.

Reply:
So a swift enum with a associated type could work here

Something like this:
``` 
enum JavaAnnotation {
  case override
  case nonnull
  .....
  case custom(name: String, fromPackage: String) // Needed for imports? 
}
```

Then the ones passed in via the CLI can be represented using the `custom` case

Before working on a PR for this would you mind putting up an RFC around the use case and general approach you'd like to take? I have some thoughts on adding more customization points to Plank and we might be able to collaborate on a solution

Reply:
Sure, I'll sync up with you on custom annotations before I add any code for that.

In the meantime, I've switched to an enum to lock it down to allowed annotations as you suggested.